src/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -168,27 +173,48 @@ public String getDescription() {
       // A small warning so users will not over-write a file,
       if (f.exists()) {
         final int choice =
-            JOptionPane.showConfirmDialog(null, "A file by that name already exists. Do you wish to over write it?",
+            JOptionPane.showConfirmDialog(frame, "A file by that name already exists. Do you wish to over write it?",
                 "Over-write?", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
         if (choice != JOptionPane.OK_OPTION) {
           return;
         }
       }
+
       final File file = f;
-      final Runnable t = () -> {
-        if (saveScreenshot(node, file, frame, gameData)) {
-          JOptionPane.showMessageDialog(null, "Map Snapshot Saved", "Map Snapshot Saved",
-              JOptionPane.INFORMATION_MESSAGE);
+      final SwingWorker<?, ?> task = new SwingWorker<Void, Void>() {
+        @Override
+        protected Void doInBackground() throws Exception {
+          saveScreenshot(node, file, frame, gameData);
+          return null;
+        }
+
+        @Override
+        protected void done() {
+          try {
+            get();
+            SwingUtilities.invokeLater(() -> {
+              JOptionPane.showMessageDialog(frame, "Map Screenshot Saved", "Map Screenshot Saved",
+                  JOptionPane.INFORMATION_MESSAGE);
+            });
+          } catch (final ExecutionException e) {
+            SwingUtilities.invokeLater(() -> {
+              JOptionPane.showMessageDialog(frame, e.getCause().getMessage(), "Error Saving Screenshot",
+                  JOptionPane.OK_OPTION);
+            });
+          } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
IMO to get the most consistent behavior, log the interrupted exception and return. Add a TODO with a link to the Goetz article, we very well may want to consider interrupting the current thread. If we really want to follow up a github issue would be a good way to track it. This should allow you to proceed while we research differences in system behavior on interrupt vs not.